components/context/ThemeContext.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import React, { createContext, useState, useContext, useEffect, ReactNode } from 'react';
+
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const storedTheme = localStorage.getItem('theme');
+  const [theme, setTheme] = useState<'dark' | 'light'>(storedTheme === 'light' ? 'light' : 'dark');
```

Comment:
I think you will face issues here, because useState will initialize the theme before localStorage has the time to get the value from storage. 
Lets try adding the  `localStorage.getItem('theme')`; inside the useState and  then you will r`eturn storedTheme === 'light' ? 'light' : 'dark'`

Reply:
Ok i changed the code.

Reply:
@DimitarHristovski awesome! 🚀  That fixed the issue of returning to the default value when reloading.

What I am noticing now:
- When the toggle is on the left(the sun side) it should be the light theme. Right now they are mixed. 
**Right side=moon side is the dark theme(the default one). Left side=sun side is the light theme.**

-Another issue we are facing is with the logo. When the user toggles the circle to the left the logo blends with the background because the word Learnhub is white like the background. 
Could you change the logo/ logo letters to be dark when we are in light theme?
Thank you 🙏

Reply:
The logo is a image that was provided before was text... i thought someone will give me the dark logo image.And about the toggle i was just following the design.If its on the right the will be dark and on the left light.